src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/AcknowledgeServiceCallbackHandler.java

```diff
@@ -66,17 +68,17 @@ private CallbackResponse validateDateOfBirth(CallbackParams callbackParams) {
     }
 
     private CallbackResponse setNewResponseDeadline(CallbackParams callbackParams) {
-        Map<String, Object> data = callbackParams.getRequest().getCaseDetails().getData();
-        LocalDateTime responseDeadline = callbackParams.getCaseData().getRespondentSolicitor1ResponseDeadline();
-
+        CaseData caseData = callbackParams.getCaseData();
+        LocalDateTime responseDeadline = caseData.getRespondentSolicitor1ResponseDeadline();
         LocalDate newResponseDate = workingDayIndicator.getNextWorkingDay(responseDeadline.plusDays(14).toLocalDate());
+        CaseData.CaseDataBuilder caseDataBuilder = caseData.toBuilder()
+            .respondentSolicitor1ResponseDeadline(newResponseDate.atTime(MID_NIGHT));
 
-        data.put(RESPONSE_DEADLINE, newResponseDate.atTime(MID_NIGHT));
-        List<String> errors = businessProcessService.updateBusinessProcess(data, ACKNOWLEDGE_SERVICE);
+        CaseData caseDataUpdated = businessProcessService.updateBusinessProcess(
+            caseDataBuilder.build(), ACKNOWLEDGE_SERVICE);
 
         return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(data)
-            .errors(errors)
+            .data(caseDetailsConverter.toMap(caseDataUpdated))
```

Comment:
what about having a toMap method on CaseData?

Reply:
I will prefer the domain objects plain java objects, don't want to inject `ObjectMapper` bean there.

Reply:
hmm we could have this in an interface as we have for DocmosisData however?